pkg/lockfile/parse.go

```diff
@@ -57,28 +59,14 @@ var ErrParserNotFound = errors.New("could not determine parser")
 
 type Packages []PackageDetails
 
-func toSliceOfEcosystems(ecosystemsMap map[Ecosystem]struct{}) []Ecosystem {
-	ecosystems := make([]Ecosystem, 0, len(ecosystemsMap))
-
-	for ecosystem := range ecosystemsMap {
-		if ecosystem == "" {
```

Comment:
Is this logic no longer necessary?

Reply:
I'm not sure that it was ever needed - I think it came over from `osv-detector`, and I think might have been a legacy of how I previously gathered ecosystems but then later refactored.

Either way removing it doesn't cause any test failures for both `osv-detector` and `osv-scanner` to fail, so I think it's best to purposely remove so we can ensure we have a test to cover it if it does turn out to be needed